From 8b508bb134187eaf7ea6258d8466fe9451ce075a Mon Sep 17 00:00:00 2001
From: John Cox <jc@kynesim.co.uk>
Date: Wed, 28 Sep 2022 15:29:35 +0000
Subject: [PATCH 136/151] v4l2_req: #include <stdint.h> where uint used in
 headers

Also stddef.h for size_t and add guard #ifdefs where missing
This fixes compile failures with linux headers from 5.19 and later.
---
 libavcodec/v4l2_req_devscan.h  | 2 ++
 libavcodec/v4l2_req_dmabufs.h  | 2 ++
 libavcodec/v4l2_req_utils.h    | 5 +++++
 libavcodec/v4l2_request_hevc.h | 1 +
 4 files changed, 10 insertions(+)

diff --git a/libavcodec/v4l2_req_devscan.h b/libavcodec/v4l2_req_devscan.h
index 0baef36535..956d9234f1 100644
--- a/libavcodec/v4l2_req_devscan.h
+++ b/libavcodec/v4l2_req_devscan.h
@@ -1,6 +1,8 @@
 #ifndef _DEVSCAN_H_
 #define _DEVSCAN_H_
 
+#include <stdint.h>
+
 struct devscan;
 struct decdev;
 enum v4l2_buf_type;
diff --git a/libavcodec/v4l2_req_dmabufs.h b/libavcodec/v4l2_req_dmabufs.h
index e36aa0a712..c1d3d8c8d7 100644
--- a/libavcodec/v4l2_req_dmabufs.h
+++ b/libavcodec/v4l2_req_dmabufs.h
@@ -1,6 +1,8 @@
 #ifndef DMABUFS_H
 #define DMABUFS_H
 
+#include <stddef.h>
+
 struct dmabufs_ctl;
 struct dmabuf_h;
 
diff --git a/libavcodec/v4l2_req_utils.h b/libavcodec/v4l2_req_utils.h
index cb4bd164b4..a31cc1f4ec 100644
--- a/libavcodec/v4l2_req_utils.h
+++ b/libavcodec/v4l2_req_utils.h
@@ -1,3 +1,7 @@
+#ifndef AVCODEC_V4L2_REQ_UTILS_H
+#define AVCODEC_V4L2_REQ_UTILS_H
+
+#include <stdint.h>
 #include "libavutil/log.h"
 
 #define request_log(...) av_log(NULL, AV_LOG_INFO, __VA_ARGS__)
@@ -20,3 +24,4 @@ static inline const char * strfourcc(char tbuf[5], uint32_t fcc) {
     return tbuf;
 }
 
+#endif
diff --git a/libavcodec/v4l2_request_hevc.h b/libavcodec/v4l2_request_hevc.h
index 830444bf92..99c90064ea 100644
--- a/libavcodec/v4l2_request_hevc.h
+++ b/libavcodec/v4l2_request_hevc.h
@@ -1,6 +1,7 @@
 #ifndef AVCODEC_V4L2_REQUEST_HEVC_H
 #define AVCODEC_V4L2_REQUEST_HEVC_H
 
+#include <stdint.h>
 #include <drm_fourcc.h>
 #include "v4l2_req_decode_q.h"
 
-- 
2.38.1

